sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        address::{AccountAddress, Bech32Address},
+        context_input::{CommitmentContextInput, ContextInput},
+        output::{unlock_condition::AddressUnlockCondition, DelegationId, DelegationOutputBuilder},
+    },
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet},
+};
+
+/// Params for `create_delegation_output()`
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateDelegationParams {
+    /// Bech32 encoded address which will control the delegation.
+    /// By default, the ed25519 wallet address will be used.
+    // TODO: https://github.com/iotaledger/iota-sdk/issues/1888
+    pub address: Option<Bech32Address>,
+    /// The amount to delegate.
+    pub delegated_amount: u64,
```

Comment:
IMO it's a tiny bit redundant to have `delegated_` here as what we primarily set is the amount and then delegated_amount is set to the same, and it's obviously delegated since it's in this struct

Reply:
I felt that it was clearer to indicate it this way, since I think of it more like we set `amount` to equal `delegated_amount` rather than the other way around

Reply:
The opposite is closer to the TIP

Reply:
But the amount field can change whereas the delegation amount cannot. It just makes more sense to say that you're setting the delegation amount which implies the amount and later you might reduce the amount

Reply:
I don't necessarily think it makes more sense and I prefer us strictly following the TIP but alright if you prefer